### PR TITLE
Fix repo name mismatch when GitHub repo name contains spaces

### LINKED ISSUE #1336: repo name mismatch with GitHub when using spaces
**Describe the bug**
When creating a GitHub repo from the Dyad desktop app, if the repo name entered includes spaces, Dyad sends the request to GitHub as-is. GitHub automatically converts spaces into hyphens (`-`) in repo names, but Dyad continues to reference the repo name with spaces. This mismatch causes Dyad to think the repo name contains spaces, while GitHub actually uses hyphens.

**To Reproduce**

1. Open the Dyad desktop app.
2. Create a new project and choose the option to create a GitHub repository.
3. Enter a repository name with a space (e.g., `my app`).
4. Submit the creation request.
5. GitHub creates a repo named `my-app`, but Dyad registers it as `my app`.
6. Observe that Dyad cannot properly match or link to the created GitHub repo.

**Expected behavior**
Dyad should sanitize the repo name before sending the request to GitHub (e.g., replace spaces with `-`) and store the normalized name (`my-app`) so that it matches the actual repo name created by GitHub.

**Screenshots**

Image

Image

Image

**Desktop (please complete the following information):**
* OS: Windows 11
* App: Dyad (Electron)
* Version: 0.21.0

**Additional context**
This issue only occurs when the user accidentally includes spaces in the repo name. Since GitHub automatically normalizes spaces to hyphens

### PR DESCRIPTION
**Pull request by wwwillchen**
## Summary
- Adds `normalizeGitHubRepoName` function that converts spaces to hyphens to match GitHub's automatic normalization behavior
- Applies normalization when creating repos, checking availability, and storing repo names in the database
- Fixes issue where Dyad stored "my app" but GitHub created "my-app", causing a mismatch

Fixes #1336

## Test plan
- Unit tests added for `normalizeGitHubRepoName` function covering edge cases
- E2E test added to verify repo names with spaces are properly normalized when creating and syncing

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

<a href="https://app.devin.ai/review/dyad-sh/dyad/pull/2377">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

---
## Summary by cubic
Normalize GitHub repo names by converting all whitespace to hyphens so our stored name matches GitHub’s actual repo name. Fixes #1336.

- **Bug Fixes**
  - Add normalizeGitHubRepoName to convert whitespace to hyphens.
  - Apply normalization for availability checks, repo creation, remote URLs, and DB storage.
  - Add unit tests for edge cases and an E2E test for create/sync with spaced names.

<sup>Written for commit 0735ce38a11fced50e670821e58289dcc53fe86f. Summary will update on new commits.</sup>

---

> [!NOTE]
> **Medium Risk**
> Touches the GitHub create/availability flow and the repo name stored in the app DB, which could break GitHub linking if normalization is incorrect. Logic is small and well-covered by new unit and E2E tests, reducing risk.
> 
> **Overview**
> Fixes repo-name mismatches when users enter names containing whitespace by introducing `normalizeGitHubRepoName` (trim + collapse whitespace to `-`).
> 
> Applies normalization to repo availability checks, repo creation payloads, generated remote URLs, and `updateAppGithubRepo` persistence so the app consistently uses GitHub’s normalized repo name.
> 
> Adds Vitest unit coverage for normalization edge cases and a Playwright E2E test that creates/syncs a repo named with spaces and asserts the hyphenated name is used end-to-end.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0735ce38a11fced50e670821e58289dcc53fe86f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by wwwillchen** [comment]
@BugBot run

File: src/ipc/handlers/github_handlers.ts [hunk 1]
```diff
@@ -689,8 +706,8 @@ async function handleCreateRepo(
 
   // Set up remote URL before preparing branch
   const remoteUrl = IS_TEST_BUILD
-    ? `${GITHUB_GIT_BASE}/${owner}/${repo}.git`
-    : `https://${accessToken}:x-oauth-basic@github.com/${owner}/${repo}.git`;
+    ? `${GITHUB_GIT_BASE}/${owner}/${normalizedRepo}.git`
+    : `https://${accessToken}:x-oauth-basic@github.com/${owner}/${normalizedRepo}.git`;
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/ipc/handlers/github_handlers.ts:747]: @/private/tmp/claude/-Users-will-Documents-GitHub-dyad/9f45c12b-86da-410d-8540-b793eebe5a77/scratchpad/comment.txt

[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/ipc/handlers/github_handlers.ts:747]: -

[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/ipc/handlers/github_handlers.ts:747]: 🚩 **Flagged for human review**: The token-in-URL pattern for git authentication is a pre-existing pattern used throughout the codebase (not introduced by this PR). Addressing this would require architectural changes to the git authentication mechanism, which is beyond the scope of this bugfix PR. The security risk is mitigated since:
1. This is an Electron app running locally
2. Error messages are logged to local electron-log files with limited exposure
3. The token has a controlled scope (repo,user,workflow)

Consider creating a separate security improvement issue to investigate using a git credential helper instead.

**Comment by wwwillchen** [comment]
@BugBot run

**Comment by wwwillchen** [comment]
@BugBot run
